internal/auth/auth.go

```diff
@@ -1,44 +1,70 @@
 package auth
 
 import (
-	"fmt"
-	"github.com/TheDevExperiment/server/internal/db/repositories"
-	"github.com/TheDevExperiment/server/router/models"
-	"github.com/gin-gonic/gin"
-	"go.mongodb.org/mongo-driver/bson"
+	"log"
 	"net/http"
+
+	"github.com/TheDevExperiment/server/internal/db/users"
+	userModels "github.com/TheDevExperiment/server/internal/db/users/models"
+	"github.com/TheDevExperiment/server/middlewares"
+	"github.com/TheDevExperiment/server/router/models/auth"
+	"github.com/gin-gonic/gin"
 )
 
 func GuestValidateV1(c *gin.Context) {
-	// first bind the req to our model
-	var req models.AuthRequest
-	var res models.AuthResponse
-	userRepository := c.MustGet("userRepository").(*repositories.UserRepository)
+	var res auth.AuthResponse
+	userRepository := users.NewUserRepository()
 
-	if err := c.ShouldBindJSON(&req); err != nil {
-		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+	//set by the middleware
+	userId := c.GetString(middlewares.ContextKeyUserId)
+	if userId == "" {
+		res.Message = http.StatusText(http.StatusUnauthorized)
+		c.JSON(http.StatusUnauthorized, res)
 		return
 	}
-
-	// if is not a guest account
-	if !req.IsGuest {
-		res.Message = "Only Guest Accounts are supported"
-		c.JSON(http.StatusNotImplemented, res)
+	data, err := userRepository.FindById(c, userId)
+	if err != nil {
+		res.Message = err.Error()
+		c.JSON(http.StatusInternalServerError, res)
 		return
 	}
-	if req.UserId == "" || req.SecretToken == "" {
-		res.Message = "UserId and SecretToken must be provided."
-		c.JSON(http.StatusBadRequest, res)
+	if data == nil {
+		res.Message = http.StatusText(http.StatusUnauthorized)
+		c.JSON(http.StatusUnauthorized, res)
 		return
 	}
+	res.Message = "Authentication Successful"
+	res.Data = append(res.Data, (*data))
+	c.JSON(http.StatusOK, res)
+}
+
+func CreateGuestV1(c *gin.Context) {
+	// first bind the req to our model
+	var req auth.CreateAccountRequest
+	var res auth.AuthResponse
```

Comment:
lets create a auth.CreateAccountResponse to avoid confusion?

Reply:
the response object is used by guest-validate route also

Reply:
added a seperate struct for createAccountResponse